cmd/dogstatsd/main.go

```diff
@@ -52,14 +71,33 @@ on dashboards. DogStatsD implements the StatsD protocol, along with a few
 extensions for special Datadog features.`,
 	}
 
-	startCmd = &cobra.Command{
+	for _, cmd := range makeCommands() {
+		dogstatsdCmd.AddCommand(cmd)
+	}
+
+	return dogstatsdCmd
+}
+
+func makeCommands() []*cobra.Command {
+	cliParams := &cliParams{}
+	startCmd := &cobra.Command{
 		Use:   "start",
 		Short: "Start DogStatsD",
 		Long:  `Runs DogStatsD in the foreground`,
-		RunE:  start,
+		RunE: func(*cobra.Command, []string) error {
+			return fxutil.OneShot(start,
+				fx.Supply(cliParams),
+				fx.Supply(core.BundleParams{
+					ConfFilePath:      cliParams.confPath,
+					ConfigLoadSecrets: false,
+					ConfigMissingOK:   true,
```

Comment:
@djmitche : Are these previous value correct (`ConfigLoadSecrets`, `ConfigMissingOK`)?

Reply:
That's a good question.  You'd need to look at the implementation of `runAgent`, below, and `pkgconfig.Load()`, to see how it behaved before.  Note that you may also need to change `core.BundleParams#ConfigName`.

If there's other required functionality that's missing in comp/core/config, please feel invited to add it!